mungegithub/mungers/milestone-maintainer_test.go

```diff
@@ -111,177 +112,380 @@ func TestMilestoneMaintainer(t *testing.T) {
+			mode:             milestoneModeSlush,
+			labelsComplete:   true,
+			isApproved:       true,
+			isBlocker:        true,
+			events:           milestoneLabelEvents(statusNeedsApprovalLabel, createdPastApprovalGracePeriod),
+			comments:         milestoneIssueComments(createdNow),
+			expectedSections: sets.NewString("summarizeLabels", "warnMissingStatus", "warnUpdateInterval"),
+			expectedState:    milestoneNeedsAttention,
+			expectedBody: `
+**Action required**: Must specify at most one of ` + "`status/in-progress` or `status/in-review`." + `
+
+**Note**: This issue is marked as ` + "`priority/critical-urgent`" + `, and must be updated every 3 days.
+
+Example update:
+
+` + "```" + `
+ACK.  In progress
+ETA: DD/MM/YYYY
+Risks: Complicated fix required
+` + "```" + `
+<details>` + blockerCompleteLabels,
+		},
+		"slush - Complete labels, approved, blocker, update not due": {
+			mode:             milestoneModeSlush,
+			labels:           []string{"status/in-progress"},
+			labelsComplete:   true,
+			isApproved:       true,
+			isBlocker:        true,
+			events:           milestoneLabelEvents(statusNeedsApprovalLabel, createdPastApprovalGracePeriod),
+			comments:         milestoneIssueComments(createdNow),
+			expectedSections: sets.NewString("summarizeLabels", "warnUpdateInterval"),
+			expectedState:    milestoneCurrent,
+			expectedBody: `
+**Note**: This issue is marked as ` + "`priority/critical-urgent`" + `, and must be updated every 3 days.
+
+Example update:
+
+` + "```" + `
+ACK.  In progress
+ETA: DD/MM/YYYY
+Risks: Complicated fix required
+` + "```" + `
+<details open>` + blockerCompleteLabels,
+		},
+		"slush - Complete labels, approved, blocker, update due": {
+			mode:             milestoneModeSlush,
+			labels:           []string{"status/in-progress"},
+			labelsComplete:   true,
+			isApproved:       true,
+			isBlocker:        true,
+			events:           milestoneLabelEvents(statusNeedsApprovalLabel, createdPastApprovalGracePeriod),
+			comments:         milestoneIssueComments(createdPastSlushUpdateInterval),
+			expectedSections: sets.NewString("summarizeLabels", "warnUpdateRequired"),
+			expectedState:    milestoneNeedsAttention,
+			expectedBody: `
+**Action Required**: This issue is marked as ` + "`priority/critical-urgent`" + `, but has not been updated in 3 days. Please provide an update.
+
+Example update:
+
+` + "```" + `
+ACK.  In progress
+ETA: DD/MM/YYYY
+Risks: Complicated fix required
+` + "```" + `
+<details>` + blockerCompleteLabels,
+		},
+		"freeze - Complete labels, approved, non-blocker": {
+			mode:             milestoneModeFreeze,
+			labelsComplete:   true,
+			isApproved:       true,
+			expectedSections: sets.NewString("summarizeLabels", "removeNonBlocker"),
+			expectedState:    milestoneNeedsRemoval,
+			expectedBody:     "**Important**: Code freeze is in effect and only issues with `priority/critical-urgent` may remain in the v1.8 milestone.",
 		},
 	}
```

Comment:
Please add a few more test cases. Specifically there are no tests for incomplete labels or unapproved issues occurring during slush or freeze.

Reply:
Behavior for those cases is the same across modes, so no extra testing should be required.  I'll add a comment to that effect.

Reply:
I've removed the `dev - ` prefix for tests that apply to all 3 modes, is that better?

Reply:
Yeah thats makes sense.
/lgtm
I'll check out your branch and test these changes in dry-run mode shortly.